lib/pacemaker/pcmk_graph_producer.c

```diff
@@ -182,119 +182,131 @@ add_downed_nodes(xmlNode *xml, const pe_action_t *action,
     }
 }
 
+/*!
+ * \internal
+ * \brief Create the transition graph XML for a scheduled action
+ *
+ * \param[in] action        Scheduled action
+ * \param[in] skip_details  If false, add action details as sub-elements
+ * \param[in] data_set      Cluster working set
+ *
+ * \return Transition graph XML for scheduled action
+ */
 static xmlNode *
-action2xml(pe_action_t * action, gboolean as_input, pe_working_set_t *data_set)
+action2xml(pe_action_t *action, bool skip_details, pe_working_set_t *data_set)
 {
-    gboolean needs_node_info = TRUE;
-    gboolean needs_maintenance_info = FALSE;
+    bool needs_node_info = true;
+    bool needs_maintenance_info = false;
     xmlNode *action_xml = NULL;
     xmlNode *args_xml = NULL;
 #if ENABLE_VERSIONED_ATTRS
     pe_rsc_action_details_t *rsc_details = NULL;
 #endif
 
-    if (action == NULL) {
+    if ((action == NULL) || (data_set == NULL)) {
         return NULL;
     }
 
+    // Create the top-level element based on task
+
     if (pcmk__str_eq(action->task, CRM_OP_FENCE, pcmk__str_casei)) {
         /* All fences need node info; guest node fences are pseudo-events */
         action_xml = create_xml_node(NULL,
                                      pcmk_is_set(action->flags, pe_action_pseudo)?
                                      XML_GRAPH_TAG_PSEUDO_EVENT :
                                      XML_GRAPH_TAG_CRM_EVENT);
 
-    } else if (pcmk__str_eq(action->task, CRM_OP_SHUTDOWN, pcmk__str_casei)) {
+    } else if (pcmk__str_any_of(action->task,
+                                CRM_OP_SHUTDOWN,
+                                CRM_OP_CLEAR_FAILCOUNT,
+                                CRM_OP_LRM_REFRESH, NULL)) {
         action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT);
 
-    } else if (pcmk__str_eq(action->task, CRM_OP_CLEAR_FAILCOUNT, pcmk__str_casei)) {
-        action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT);
-
-    } else if (pcmk__str_eq(action->task, CRM_OP_LRM_REFRESH, pcmk__str_casei)) {
-        action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT);
-
-    } else if (pcmk__str_eq(action->task, CRM_OP_LRM_DELETE, pcmk__str_casei)) {
+    } else if (pcmk__str_eq(action->task, CRM_OP_LRM_DELETE, pcmk__str_none)) {
         // CIB-only clean-up for shutdown locks
         action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT);
         crm_xml_add(action_xml, PCMK__XA_MODE, XML_TAG_CIB);
 
-/* 	} else if(pcmk__str_eq(action->task, RSC_PROBED, pcmk__str_casei)) { */
-/* 		action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT); */
+#if 0
+    } else if (pcmk__str_eq(action->task, RSC_PROBED, pcmk__str_none)) {
+        action_xml = create_xml_node(NULL, XML_GRAPH_TAG_CRM_EVENT);
+#endif
```

Comment:
For blocks like this, I'd love a comment indicating why it is not currently enabled.

Reply:
Agreed, added commit on top cleaning up the situation